great_expectations/checkpoint/actions.py

```diff
@@ -147,6 +137,16 @@ def _get_data_docs_pages_from_prior_action(
         return None
 
 
+def _is_enabled(success: bool, notify_on: Literal["all", "failure", "success"]) -> bool:
+    return (
+        notify_on == "all"
+        or notify_on == "success"
+        and success
+        or notify_on == "failure"
+        and not success
+    )
```

Comment:
I don't love it but a shared utility is the easiest way to get this functional - could subclass for conditional actions but that feels drastic for what should be a very small change